src/channel/ably/utils.ts

```diff
@@ -0,0 +1,174 @@
+export const isNullOrUndefined = (obj) => obj == null || obj === undefined;
+export const isEmptyString = (stringToCheck, ignoreSpaces = true) => (ignoreSpaces ? stringToCheck.trim() : stringToCheck) === '';
+export const isNullOrUndefinedOrEmpty = (obj) => obj == null || obj === undefined || isEmptyString(obj);
+
+/**
+ * @throws Exception if parsing error
+ */
+export const parseJwt = (jwtToken: string): { header: any, payload: any } => {
+    // Get Token Header
+    const base64HeaderUrl = jwtToken.split('.')[0];
+    const base64Header = base64HeaderUrl.replace('-', '+').replace('_', '/');
+    const header = JSON.parse(toText(base64Header));
+    // Get Token payload
+    const base64Url = jwtToken.split('.')[1];
+    const base64 = base64Url.replace('-', '+').replace('_', '/');
+    const payload = JSON.parse(toText(base64));
+    return { header, payload };
+}
+
+export const toTokenDetails = (jwtToken: string) => {
+    const { payload } = parseJwt(jwtToken);
+    return {
+        // capability: parsedJwt['x-ably-capability'], // RSA4f - tokenDetails size should't exceed 128kb
```

Comment:
Probably can be removed

Reply:
Yeah ... but I am afraid someone in the future might try to add it, I will remove `capability` section and keep the comment as is!

Reply:
Fixed!